Tests/FoundationEssentialsTests/SubprocessTests.swift

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if FOUNDATION_FRAMEWORK
+@testable import Foundation
+#else
+@testable import FoundationEssentials
+#endif // FOUNDATION_FRAMEWORK
+
+#if canImport(TestSupport)
+import TestSupport
+#endif
+
+final class SubprocessTests: XCTestCase {
+    func testSimple() async throws {
+        let ls = try await Subprocess.run(executing: .named("ls"), output: .collect, error: .discard)
+        let result = String(data: ls.standardOutput!, encoding: .utf8)!
+        XCTAssert(ls.terminationStatus.isSuccess)
+        XCTAssert(!result.isEmpty)
+    }
+
+    func testComplex() async throws {
+        struct Address: Codable {
+            let ip: String
+        }
+
+        let result = try await Subprocess.run(
+            executing: .named("curl"),
+            arguments: ["http://ip.jsontest.com/"]
```

Comment:
Is it ok for our test to rely on this external address, or should we use something like `apple.com`?